### PR TITLE
`Modal`/` Flyout` - Add `returnFocusTo` documentation

### PR DESCRIPTION
**Pull request by didoo**
### :pushpin: Summary

Follow-up of https://github.com/hashicorp/design-system/pull/2497, to add the missing documentation.

### :hammer_and_wrench: Detailed description

In this PR I have:
- updated `returnFocusTo` to Component API documentation for `Modal` and `Flyout`
- added info about `returnFocusTo` in “Focus and focus trap” sections of the `Modal` and `Flyout` documentation pages

***

### 👀 Component checklist

:speech_balloon: Please consider using [conventional comments](https://conventionalcomments.org/) when reviewing this PR.

### CODE REVIEW

File: website/docs/components/flyout/partials/code/component-api.md [hunk 1]
```diff
@@ -18,6 +18,9 @@
   <C.Property @name="size" @type="enum" @values={{array "medium" "large" }} @default="medium">
     Sets the width of the Flyout.
   </C.Property>
+  <C.Property @name="returnFocusTo" @type="string">
```
Comment (didoo) [inline website/docs/components/flyout/partials/code/component-api.md:21]: Let me know if you think I should add also an example in the "how to use" section.

File: website/docs/components/modal/partials/code/component-api.md [hunk 2]
```diff
@@ -18,6 +18,9 @@
   <C.Property @name="color" @type="enum" @values={{array "neutral" "warning" "critical" }} @default="neutral">
     Sets the color scheme for the Modal Header elements: icon, tagline, and title.
   </C.Property>
+  <C.Property @name="returnFocusTo" @type="string">
```
Comment (didoo) [inline website/docs/components/modal/partials/code/component-api.md:21]: Let me know if you think I should add also an example in the "how to use" section.

File: website/docs/components/flyout/partials/code/component-api.md (hunk 1 above)
Comment (zamoore) [reply website/docs/components/flyout/partials/code/component-api.md:21]: I think it's a good idea, but not a blocker

**Review by zamoore — APPROVED** [review]

File: website/docs/components/modal/partials/code/component-api.md [hunk 3]
```diff
@@ -18,6 +18,9 @@
...
+    The `id` of the element where the focus should be returned once the modal is closed (eg. if the element that initiated the event is not in the DOM anymore once closed).
```
Comment (KristinLBradley) [inline website/docs/components/modal/partials/code/component-api.md:22]:
```suggestion
    The `id` of the element where the focus should be returned once the modal is closed (e.g., if the element that initiated the event is not in the DOM anymore once closed).
```

**Review by KristinLBradley — APPROVED** [review]

File: website/docs/components/flyout/partials/code/component-api.md (hunk 1 above)
Comment (didoo) [reply website/docs/components/flyout/partials/code/component-api.md:21]: Done. Can you check the updates? thanks

**Review by MelSumner — APPROVED** [review]

**Pull request merged**